cpp/src/interop/from_arrow_device.cu

```diff
@@ -158,12 +155,15 @@ dispatch_tuple_t dispatch_from_arrow_device::operator()<cudf::string_view>(
       {});
   }
 
-  auto offsets_view = column_view{data_type(type_id::INT32),
-                                  static_cast<size_type>(input->offset + input->length) + 1,
+  data_type offsets_type(type_id::INT32);
+  if (schema->type == NANOARROW_TYPE_LARGE_STRING) { offsets_type = data_type(type_id::INT64); }
+  auto offsets_view = column_view{offsets_type,
+                                  static_cast<int32_t>(input->offset + input->length) + 1,
```

Comment:
This cast looks incorrect if the offsets are int64

Reply:
that cast is for the length of the column, not the offset values. `cudf::column_view` expects `size_type` for the length, which is an `int32_t`

Reply:
Ah yes. Can we change the cast to `static_cast<size_type>`?